pkg/diffs/diffs.go

```diff
@@ -217,3 +217,39 @@ func getTestsByName(tests []cioperatorapi.TestStepConfiguration) map[string]ciop
 	}
 	return ret
 }
+
+// GetPresubmitsForClusterProfiles returns a filtered list of jobs from the
+// Prow configuration, with only presubmits that use certain cluster profiles.
+func GetPresubmitsForClusterProfiles(prowConfig *prowconfig.Config, profiles []config.ClusterProfile, logger *logrus.Entry) config.Presubmits {
+	matches := func(job *prowconfig.Presubmit) bool {
+		if job.Agent != string(pjapi.KubernetesAgent) {
+			return false
+		}
+		for _, v := range job.Spec.Volumes {
+			if v.Name != "cluster-profile" || v.Projected == nil {
+				continue
+			}
+			for _, s := range v.Projected.Sources {
+				if s.ConfigMap == nil {
+					continue
+				}
+				n := strings.TrimPrefix(s.ConfigMap.Name, "cluster-profile-")
+				for _, p := range profiles {
+					if n == p.Name {
```

Comment:
since we need only the names, why not passing a `sets.String` instead of `[]config.ClusterProfile`

Reply:
It would complicate the caller, give us only a very minimal performance gain (1/15th of the total execution time of the function, in my most pessimistic and unrealistic profile) and not make the code much clearer, IMO.